FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -247,6 +247,13 @@ - (void)tokenWithAuthorizedEntity:(NSString *)authorizedEntity
           return;
         }
 
+        if (!self) {
+          NSError *noSelf = [NSError messagingErrorWithCode:kFIRMessagingErrorCodeUnknown
```

Comment:
nit: maybe just error or referenceError

Reply:
Also we should give a new error code than kFIRMessagingErrorCodeUnknown

Reply:
Fixed this - renamed to lostRefError.

Reply:
@charlotteliang - 
I was torn about the error code for the following reasons -
- All existing (other) error codes we have are about Firebase (FCM) related errors. Self reference lost is not specific to Firebase and can occur in any kind of app. So didn't want to add a generic unrelated error code to Firebase set of error codes.
- We don't know at this place in code why the self reference was lost (i.e. why was TokenManager dealloced). 

Due to these reasons, I chose to leave error code as Unknown but provided a descriptive error message (failureReason) so developers can understand what is going on from the error message / logs.

Reply:
you can create a error code for all self reference errors. I wonder if there's another error you can reuse (e.g. internal error) other than Unknown.

Reply:
There isn't an internal error code. I can create one - maybe it will be available for such scenarios in the future.

Reply:
Added an internal code instead of unknown.